gilbreth_perception/src/recognition_node.cpp

```diff
@@ -384,6 +387,22 @@ class RecognitionClass {
       pick_point.z = pick_pose[result.item_id][2];
       pick_point_cloud->push_back(pick_point);
       pcl::transformPointCloud(*pick_point_cloud, *rotated_pick_point_cloud, result.final_transformation);
+      // Use ICP to fine align model to scene
+      t_start = std::clock();
```

Comment:
Why would you need to use ICP if you had already ran Correspondence Grouping?  Shouldn't you be able to use the transform from correspondence grouping to align the point cloud?

Reply:
Correspondence grouping is kind of initial alignment, sometimes it is not very accurate, so the fine alignment step is needed.

Reply:
Would it make more sense then to skip correspondence grouping and run ICP only?

Reply:
I think I tried only ICP at the very beginning of this project, but the recognition result is not good, the fitness score was not a good metric for recognition, for example, there is a gear in the scene, the fitness score for matching model gear to scene gear is in the order of 10^-7,  the fitness score for matching model disk to scene gear is also in the order of 10^-7, there is a high chance that the system think the gear is a disk.
Also, there might be something wrong with my previous code, I can double check when I got time.

Reply:
I see, lets use what you have right now and worry about those details later once we have running demo.